### PR TITLE
Switch to std::shared_ptr<TRI_vocbase_t> then remove DataProtector from DatabaseFeature

### PR DESCRIPTION
**Pull request by ObiWahn**
I think it is not as easy as thought. I guess this is not 100% correct at the moment.

Update:
We need to change the internal vocbase pointers to be shared pointers as well, this would take too long and maybe cause deadlocks on devel, if we do not carefully use a combination of shared and weak pointers.

### CODE REVIEW

File: arangod/RestServer/DatabaseFeature.cpp [hunk 1]
```diff
@@ -117,25 +117,22 @@ void DatabaseManagerThread::run() {
           MUTEX_LOCKER(mutexLocker, databaseFeature->_databasesMutex);
 
           // Build the new value:
-          auto oldLists = databaseFeature->_databasesLists.load();
+          auto oldLists = std::atomic_load(&databaseFeature->_databasesLists);
```
Comment (graetzer) [inline arangod/RestServer/DatabaseFeature.cpp]: this is not lock free anymore

**Comment by graetzer** [comment]
vasily tried it once already https://github.com/arangodb/arangodb/pull/8072

**Comment by mpoeter** [comment]
Just out of curiosity - what is the motivation behind this change?

BTW: I can't resist to note that this is the perfect use cast of a reclamation scheme from my xenium library (#8267) - especially if you want to achieve lock/wait-freedom. 😉

**Comment by jsteemann** [comment]
@mpoeter: the motivation for this was to get rid of a few plain pointers, however, things are much more complicated here.
We still have to look into your library. Due to lack of time this hasn't happened yet. :disappointed: 
I guess after the release of 3.5 we will have time to take a proper look.

**Comment by mpoeter** [comment]
Got it. No worries, I know that you are busy with other stuff ATM. :slightly_smiling_face: 
I might also take a look at this, so I can provide some insight from my side...

**Comment by jsteemann** [comment]
@mpoeter: thanks for the offer, but for now I think the code at that place can stay as it is now.
We may revisit it when we start developing things for 3.6, but I guess this is not going to happen in the next 3-4 weeks.

**Pull request closed without merging**